Rover: Added reversing loiter_mode

sahith22 opened the pull request:
I have added parameter 2 to loiter type. This allows the boat to face its stern towards the target and move towards it.

IamPete1 commented:
@sahith22 Are you happy for me to squash your commits for you?

amilcarlucas commented on APMrover2/mode_loiter.cpp:
```diff
@@ -43,13 +43,13 @@ void ModeLoiter::update()
         float yaw_error_cd = wrap_180_cd(_desired_yaw_cd - ahrs.yaw_sensor);
-        // if destination is behind vehicle, reverse towards it
-        if (fabsf(yaw_error_cd) > 9000 && g2.loit_type == 0) {
+        // if destination is behind vehicle, reverse towards it    
```
Whitespace at end of line

amilcarlucas commented on APMrover2/mode_loiter.cpp:
```diff
@@ -43,13 +43,13 @@ void ModeLoiter::update()
             _desired_speed = -_desired_speed;
         }
-
+        
```
Unnecessary whitespace change

amilcarlucas commented:
It can now be rebased and squashed.

IamPete1 commented:
I manged to break this while attempting to squash, Sorry! This git stuff is tricky, I have [re-opened](https://github.com/ArduPilot/ardupilot/pull/12742) it from my backup copy of this branch. I think once I had updated this to master there was no difference so the PR was closed then I was no longer allowed to push the squashed commit?

sahith22 commented:
The master does not contain the updated code.

IamPete1 commented:
maybe I should add that you can fix it with
git remote add iampete1 https://github.com/IamPete1/ardupilot.git
git fetch iampete1
git cherry-pick ed6a47e45ed54e8569f0916663b87eb4f725a59a
git push -f

sahith22 commented:
upon using git push-f , I am told that it is unable to access 'https://github.com/IamPete1/ardupilot.git/': The requested URL returned error: 403.

IamPete1 commented:
hum what does 'git status' give?

sahith22 commented:
Upon using git status it gives-
You are currently cherry-picking commit ed6a47e.
  (all conflicts fixed: run "git cherry-pick --continue")
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

Changes to be committed:

	modified:   APMrover2/Parameters.cpp
	modified:   APMrover2/mode_loiter.cpp

IamPete1 commented:
sounds like you need to do git commit

some thing like 'git commit -a -m "Rover: Added reversing loiter_type parameter" '

I can recommend using github desktop, it makes it abit easier
